tutorials/tutorial_views.py

```diff
@@ -0,0 +1,944 @@
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+import pyautogui as pag
+import multiprocessing
+import sys
+from sys import platform
+from pyscreeze import ImageNotFoundException
+from tutorials import screenrecorder as sr
+from mslib.msui import mss_pyui
+
+
+def initial_ops():
+    """
+    Executes the initial operations such as closing all opened windows and showing the desktop.
+    """
+    pag.sleep(5)
+    if platform == "linux" or platform == "linux2":
+        pag.hotkey('winleft', 'd')
+        print("\n INFO : Automation is running on Linux system..\n")
+    elif platform == "darwin":
+        pag.hotkey('option', 'command', 'm')
+        print("\n INFO : Automation is running on Mac OS..\n")
+    elif platform == "win32":
+        pag.hotkey('win', 'd')
+        print("\n INFO : Automation is running on Windows OS..\n")
+    else:
+        pag.alert(text="Sorry, no support on this platform!", title="Platform Exception", button='OK')
+    print("\n INFO : We will be learning how to create a project in MSS with all the views.\n")
+
+
+def call_recorder():
+    """
+    Calls the screen recorder class to start the recording of the automation.
+    """
+    sr.main()
+
+
+def call_mss():
+    """
+    Calls the main MSS GUI window since operations are to be performed on it only.
+    """
+    mss_pyui.main()
+
+
+def automate_views():
+    """
+    This is the main automating script of the MSS views tutorial which will cover all the views(topview, sideview,
+    tableview, linear view) in demonstrating how to create a project. This will be recorded and savedto a file having
+     dateframe nomenclature with a .mp4 extension(codec).
+    """
+    # Giving time for loading of the MSS GUI.
+    pag.sleep(5)
+
+    if platform == 'linux' or platform == 'linux2' or platform == 'darwin':
+        dir_path = 'pictures/tutorial_views/win/'
+        wms_path = 'pictures/tutorial_wms/linux/'
+    elif platform == 'win32':
+        dir_path = 'pictures/tutorial_views/win/'
+        wms_path = 'pictures/tutorial_wms/win/'
+
+    # Screen Resolutions
+    sc_width, sc_height = pag.size()[0] - 1, pag.size()[1] - 1
+
+    # Maximizing the window
+    try:
+        if platform == 'linux' or platform == 'linux2':
+            pag.hotkey('winleft', 'up')
+        elif platform == 'darwin':
+            pag.hotkey('ctrl', 'command', 'f')
+        elif platform == 'win32':
+            pag.hotkey('win', 'up')
+    except Exception:
+        print("\nException : Enable Shortcuts for your system or try again!")
+    pag.sleep(2)
+    pag.hotkey('ctrl', 'h')
+    pag.sleep(2)
+
+    # Shfting topview window to upper right corner
+    try:
```

Comment:
comment: usually we do capsulate in try exception blocks only the part of the lines which could make trouble

Reply:
I have used try except in every part where it is locating an image on the screen to click upon. If the try block would be removed and if it is not able to find the image then the user will not know where the error in program is or it would simply stop the execution.
Wherever there is `pag.locateCentreOnScreen()` function I have kept it inside the try block and all the following code that is dependent on that image found by the same function.